doc/mgmt/SONiC_AliasFeature_HLD.md

```diff
@@ -0,0 +1,411 @@
+
+# Scope
+
+Covers general design for supporting interface-naming feature. Scope of interface-naming feature is restricted to SONiC management framework. SONiC CLICK based support is unaffected. Setttig interface-naming to standard in config_db.json is mentioned in the ZTP section.
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+
+# 1 Feature Overview
+
+The document covers various interfaces for interface-naming feature using SONiC management framework. Currently in SONiC, the ethernet interfaces are shown as a flat number. This flat interface numbering does not give mapping information about actual ports that are mapped to the interface number. By setting interface-naming to standard, user has better usability & mapping for actual port numbers.
+
+## 1.1 Requirements
+
+### 1.1.1 Functional Requirements
+
+Provide ability to have an alternative configuration for Ethernet Interface using SONiC management framework.
+
+### 1.1.2 Configuration and Management Requirements
+
+- Two ways to set the `interface-naming`. They are classified as `standard` and `native`.
+- Implement interface-naming related config & show commands.
+- REST & gNMI set/get support for interface-naming.
+
+### 1.1.3 Scalability Requirements
+
+### 1.1.4 Warm Boot Requirements
+
+## 1.2 Design Overview
+
+### 1.2.1 Basic Approach
+
+- By default system will boot in native interface-naming. Native interface-naming is where interfaces are displayed as Ethernet 1, Ethernet 4, Ethernet 8 etc. This also ensures that backwards compatibility is maintained.
+- Once interface naming is set to standard, all subsequent CLI, REST & gNMI interfaces will use the alternative name for ethernet interface configuration & retrieval.
+- Existing SSH sessions:- All the existing sessions where change in interface-naming mode is required will be notified to restart their existing sonic-cli sessions. This will be supported through `wall`command.
+
+### 1.2.2 Container
+
+### 1.2.3 SAI Overview
+
+# 2 Functionality
+
+## 2.1 Target Deployment Use Cases
+
+## 2.2 Functional Description
+
+# 3 Design
+
+## 3.1 Overview
+
+- For dynamically using the right string for ethernet interfaces (i.e Ethernet 4 in native mode & Eth 1/4 in standard mode), XML file has to be modified to use the correct PTYPE. `PHY_INTERFACE` has to be used as PTYPE for Ethernet Interface.
+- When interface-naming is set to standard, ethernet interfaces will be represented as Eth[slot/port/breakout-port], where slot will start from 1 and port/breakout-port numbers will start from 1. Slot will always likely to be 1 in fixed pizza-box format, but may take different values in chassis format. This name format is fixed and will not have references to interface speed. For e.g Eth1/1 (in non breakout mode), Eth1/1/1 (in breakout mode).
+- The standard interface name {e.g Eth1/1) is picked from platform.json file. All platforms need to be updated with this format for port name.

```

Comment:
Please confirm alias  is picked up from CONFIG_DB not from platform.json, the back end application populate DB entries from platform.json based on break-out mode.

Reply:
Need updated platform.json with the new format for ports. Interface-naming API's use data from CONFIG_DB.